pkgs/http/test/io/request_test.dart

```diff
@@ -46,15 +46,17 @@ void main() {
     final response = await request.send();
 
     expect(response.statusCode, equals(302));
+    expect(
+        (response as http.BaseResponseV2).url, serverUrl.resolve('/redirect'));
```

Comment:
[nit] Prefer to encode the type expectation with a matcher instead of casting and relying on the cast failure to fail the test.

```suggestion
      expect(
          response,
          isA<http.BaseResponseV2>()
              .having((r) => r.url, 'url', serverUrl.resolve('/redirect')));
```

Reply:
Done